### PR TITLE
Fix authentication error for client-auth-plugin

### LINKED ISSUE #240: client-keystone-auth: Project name is missing for authentication
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
client-keystone-auth doesn't work

**What you expected to happen**:
client-keystone-auth should work after kubeconfig configuration.

**How to reproduce it (as minimally and precisely as possible)**:
1. `kubectl config set-credentials openstackuser-clientauth`
2. Edit ~/.kube/config file, add the information client-keystone-auth needs

```yaml
users:
- name: openstackuser-clientauth
  user:
    exec:
      command: "/home/ubuntu/client-keystone-auth"
      apiVersion: "client.authentication.k8s.io/v1alpha1"
      env:
        - name: "OS_USERNAME"
          value: "demo"
        - name: "OS_PASSWORD"
          value: "password"
        - name: "OS_PROJECT_NAME"
          value: "demo"
        - name: "OS_TENANT_NAME"
          value: "demo"
        - name: "OS_REGION_NAME"
          value: "RegionOne"
        - name: "OS_IDENTITY_API_VERSION"
          value: "3"
        - name: "OS_DOMAIN_NAME"
          value: "default"
      args:
        - "--keystone-url=http://10.140.81.86/identity/v3"
        - "--domain-name=default"
```

3. In webhook service policy definition, allow user `demo` with `kube-viewer` role to get pods
4. Run `kubectl get pods`, the authorization failed.

```shell
$ kubectl get po
Keystone auth url is "http://10.140.81.86/identity/v3"
Domain name is "default"
User name is "demo"
Password is set
Error from server (Forbidden): pods is forbidden: User "demo" cannot list pods in the namespace "default"
```

The reason is the token client-keystone-auth gets from keystone is an unscoped token because the project name is not specified when fetching token.

### PR DESCRIPTION
**Pull request by lingxiankong**
**What this PR does / why we need it**:
Fix authentication error for client-auth-plugin

**Which issue this PR fixes**: 
fixes #240

**Special notes for your reviewer**:
With this patch, the client-keystone-auth could work the same with the traditional way. You can config the kubeconfig file as the following, you can use environment variables and don't need to add any params:

```yaml
users:
- name: openstackuser-clientauth
  user:
    exec:
      command: "/home/ubuntu/client-keystone-auth"
      apiVersion: "client.authentication.k8s.io/v1alpha1"
```

Then run `kubectl get pods`.

**Release note**:

```release-note
```

### CODE REVIEW

File: cmd/client-keystone-auth/main.go [hunk 1]
```diff
@@ -150,13 +150,15 @@ func main() {
...
 	var user string
+	var project string
 	var password string
 	var options gophercloud.AuthOptions
 	var err error
 
 	pflag.StringVar(&url, "keystone-url", os.Getenv("OS_AUTH_URL"), "URL for the OpenStack Keystone API")
 	pflag.StringVar(&domain, "domain-name", os.Getenv("OS_DOMAIN_NAME"), "Keystone domain name")
 	pflag.StringVar(&user, "user-name", os.Getenv("OS_USERNAME"), "User name")
+	pflag.StringVar(&project, "project-name", os.Getenv("OS_PROJECT_NAME"), "Keystone project name")
```
Comment (oomichi) [inline cmd/client-keystone-auth/main.go]: The environment variable OS_PROJECT_NAME is valid for the project name in OpenStack as http://git.openstack.org/cgit/openstack/python-openstackclient/tree/README.rst#n85 so LGTM

File: cmd/client-keystone-auth/main.go [hunk 2]
```diff
@@ -84,40 +84,40 @@ func prompt(url string, domain string, user string, password string) (gopherclou
 		if err != nil {
 			return options, err
 		}
-	} else {
-		fmt.Fprintf(os.Stderr, "Keystone auth url is \"%s\"\n", url)
```
Comment (oomichi) [inline cmd/client-keystone-auth/main.go]: According to the reported issue, the above message in the console seems a little odd, so it is nice to remove the line in this PR,

**Comment by Fedosin** [comment]
Thanks for the patch :) I wanted to do it myself could not find the time for that

**Pull request merged**